### PR TITLE
fix: contributors list

### PR DESCRIPTION
**Pull request by vasu-gupta01**
An alternate getContributors method was created in the Experiments class. The issue seems to be resolved but could use some further testing to verify that the checkboxes are not misbehaving.

### CODE REVIEW

File: app/src/main/java/com/T05/krowdtrialz/ui/contributors/ContributorsActivity.java [hunk 1]
```diff
@@ -52,28 +54,37 @@ public void onSuccess(Experiment experiment) {
...
 
             @Override
             public void onFailure() {
                 currentExperiment = null;
             }
         });
+
+    }
+
+    private void updateConList(Experiment exp){
```
Comment (arrooney) [inline app/src/main/java/com/T05/krowdtrialz/ui/contributors/ContributorsActivity.java:68]: maybe name this `updateContributorList` and add a javadoc comment

**Comment by vasu-gupta01** [comment]
@arrooney @Ricky-Au Using a button instead of checkbox seems to fix the issue. Previously, calling setChecked() was triggering the onCheckedListener causing it to misbehave.

File: app/src/main/java/com/T05/krowdtrialz/ui/contributors/ContributorList.java [hunk 2]
```diff
@@ -53,30 +54,29 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
...
+                if(experiment.isIgnored(contributor)){
+                    experiment.removeIgnoredUser(contributor);
+                    db.updateExperiment(experiment);
+                    ignoreButton.setText("Ignore");
+                    Log.d(TAG, "Un-Ignored User");
+                } else{
+                    experiment.ignoreUser(contributor);
+                    db.updateExperiment(experiment);
+                    ignoreButton.setText("Un-Ignore");
+                    Log.d(TAG, "Ignored User");
```
Comment (arrooney) [inline app/src/main/java/com/T05/krowdtrialz/ui/contributors/ContributorList.java:79]: Can you also toggle the colour here - both when the button text is first set and in the onclick. I think orange (0xFFFF9800) would be good for whenever the box says "Ignore" and Blue (0xFF3F51B5) whenever the button says "Un-ignore".

**Review by arrooney — CHANGES_REQUESTED** [review]
Seems to fix the problem -- just a few UI improvements to be made

**Review by Ricky-Au — APPROVED** [review]

**Review by arrooney — APPROVED** [review]

**Pull request merged**